test/ViewQuery/Dimensions.test.js

```diff
@@ -78,22 +61,71 @@ ex:view a view:View ;
   it(
     'gets a blank as property when view:as is not defined and path is a blank',
     () => {
-      const viewTTL = `
-@prefix view: <https://cube.link/view/> .
-@prefix ex: <http://example.org/> .
-ex:view a view:View ;
-  view:dimension [
-        view:from [
-            view:path [ ex:notMe ex:please ] ;
-        ] ;
-    ] .
-`
-      const parser = new Parser()
-      const dataset = rdf.dataset().addAll(parser.parse(viewTTL))
-      const view = clownface({ dataset, term: ns.ex.view })
+      const view = parse`
+        ${ns.ex.view} a ${ns.view.View} ;
+          ${ns.view.dimension} [
+                ${ns.view.from} [
+                    ${ns.view.path} [ ${ns.ex.notMe} ${ns.ex.please} ] ;
+                ] ;
+            ] .
+      `.node(ns.ex.view)
 
       const dimensions = new Dimensions({ view, variable: () => {} })
 
       strictEqual('BlankNode', dimensions.array[0].property.termType)
     })
+
+  it('does not set isResult flag when dimension is not used in projection', () => {
```

Comment:
This one is testing for 'set isResult flag when dimension is used in projection'

Falsifiable by the followind data

```js
    const view = parse`
      ${ns.ex.view} a ${ns.view.View} ;
        ${ns.view.projection} [
          ${ns.view.columns} ( _:dimension _:excluded ) ;
        ] ;
        ${ns.view.dimension} _:dimension, _:excluded .
        
      _:dimension ${ns.view.from} [
        ${ns.view.path} ${ns.ex.some} 
      ] .
      
      _:excluded ${ns.view.from} [
        ${ns.view.path} ${ns.ex.notThat} 
      ] .
      ```

Reply:
Good catch. This is why I prefer TypeScript. The problem was

```
-       ? projectedColumns.has(dimension)
+       ? projectedColumns.has(dimension.term)
```

Which would have been caught by the type system